kivy/uix/widget.py

```diff
@@ -316,7 +320,8 @@ class Widget(WidgetBase):
     '''
 
     __metaclass__ = WidgetMetaclass
-    __events__ = ('on_touch_down', 'on_touch_move', 'on_touch_up')
+    __events__ = ('on_touch_down', 'on_touch_move', 'on_touch_up',
+                  'on_kv_pre', 'on_kv_post')
```

Comment:
Are you planning to also implement `on_kv_applied` like we mentioned in the PR? I.e. option 1.